File: api/src/main/java/gg/mineral/api/network/connection/Connection.java
package gg.mineral.api.network.connection;

import java.util.UUID;

import dev.zerite.craftlib.chat.component.BaseChatComponent;
import gg.mineral.api.MinecraftServer;
import gg.mineral.api.entity.living.human.Player;
import gg.mineral.api.network.packet.Packet;

public interface Connection extends PreLoginConnection {
    /**
     * Disconnects the connection.
     * 
     * @param disconnectMessage
     */
    void disconnect(BaseChatComponent disconnectMessage);

    /**
     * Gets the player attached to the connection.
     * 
     * @return The player attached to the connection.
     */
    Player getPlayer();

    /**
     * Gets the UUID of the player attached to the connection.
     * 
     * @return The UUID of the player attached to the connection.
     */
    UUID getUuid();

    /**
     * Gets the server.
     * 
     * @return The server.
     */
    MinecraftServer getServer();

    /**
     * Queues packets to be sent to the client.
     * 
     * @param packets
     */
    void queuePacket(Packet.OUTGOING... packets);
}

File: api/src/main/java/gg/mineral/api/network/connection/Connection.kt
package gg.mineral.api.network.connection

import dev.zerite.craftlib.chat.component.BaseChatComponent
import gg.mineral.api.MinecraftServer
import gg.mineral.api.entity.living.human.Player
import gg.mineral.api.network.packet.Packet
import java.util.*

interface Connection : PreLoginConnection {
    /**
     * Disconnects the connection.
     *
     * @param disconnectMessage
     */
    fun disconnect(disconnectMessage: BaseChatComponent?)

    /**
     * Gets the player attached to the connection.
     *
     * @return The player attached to the connection.
     */
    val player: Player?

    /**
     * Gets the UUID of the player attached to the connection.
     *
     * @return The UUID of the player attached to the connection.
     */
    val uuid: UUID?

    /**
     * Gets the server.
     *
     * @return The server.
     */
    val server: MinecraftServer?

    /**
     * Queues packets to be sent to the client.
     *
     * @param packets
     */
    fun queuePacket(vararg packets: Packet.OUTGOING?)
}
